src/compas/geometry/primitives/polyline.py

```diff
@@ -412,4 +412,74 @@ def divide_polyline_by_length(self, length, strict=True, tol=1e-06):
 
         return division_pts
 
+    def extend(self, length, both_ends=False):
+        """Extends a polyline from its start or end by a given length
+
+        Parameters:
+        -----------
+        length: list of float [start_offset, end_offset]
+
+        both_ends: bool
+            If True, both ends of Polyline will be extended
+
+        Returns
+        -------
+        Polyline
+        """
+        if not isinstance(length, list):
+            length = [length, length]
+
```

Comment:
I would add here argument validation, because you check for specific indices later on, so perhaps, something like:
```suggestion
        length = list(length)

        if len(length) < 2:
            raise ValueError('length')
```